report_format_option/models/ir_actions_report.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 Quartile Limited
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from odoo import fields, models
+
+
+class IrActionReport(models.Model):
+    _inherit = "ir.actions.report"
+
+    encoding = fields.Char(
+        help="Encoding to be applied to the generated CSV file. e.g. cp932"
+    )
+    encode_error_handling = fields.Selection(
+        selection=[("ignore", "Ignore"), ("replace", "Replace")],
+        help="If nothing is selected, CSV export will fail with an error message when "
+        "there is a character that fail to be encoded.",
+    )
+    show_encoding = fields.Boolean(
+        compute="_compute_show_encoding",
+        help="Technical field to control the visibility of the encoding field.",
+    )
+    line_ending = fields.Char(
```

Comment:
@AungKoKoLin1997 Can we actually make it a selection field, pre-defining 'LF', 'CRLF' and 'CR'?

I think we can also extend methods `_render_qweb_html()` and `_render_qweb_text()` in this module to handle the conversion of line ending and encoding.